backend/lib/deal-observer.js

```diff
@@ -1,11 +1,35 @@
 /** @import {Queryable} from '@filecoin-station/deal-observer-db' */
 /** @import {Provider} from 'ethers' */
 
-/**
- * Observe the built-in actor events on the Filecoin blockchain to discover new FIL+ deals
- * @param {Queryable} pgPool
- * @param {Provider} rpcProvider
- */
-export const observeBuiltinActorEvents = async (pgPool, rpcProvider) => {
-  // TBD
+import { EVENT_TYPES } from './config.js'
+import { ActorEventFilter, LotusService } from './rpc-service/service.js'
+
+class DealObserver {
+  #pgPool
+  #lotusService
+  #cache
+
+  constructor (pgPool = null, chainHead = null) {
+    // TODO: Store events in pgPool
+    this.#pgPool = pgPool
+    this.#cache = new Map()
+    this.#cache.set('chainHead', chainHead)
+  }
+
+  async build () {
+    this.#lotusService = await (new LotusService()).build()
```

Comment:
Since an instance of this class cannot be used before it's built, I think it's better to create a static asynchronous factory method.

```suggestion
    this.#lotusService = await LotusService.create()
```

WDYT?

Reply:
Sounds good to me.